lib/operations/mongo_client_ops.js

```diff
@@ -194,6 +197,31 @@ function connect(mongoClient, url, options, callback) {
       delete _finalOptions.db_options.auth;
     }
 
+    // APM logging
+    if (_finalOptions.monitor) {
+      // get array of monitor groups to enable
+      const monitors = getMonitors(_finalOptions.monitor);
+      // check if logging enables command monitoring
+      if (monitors.indexOf('command') > -1) _finalOptions.monitorCommands = true;
+
+      const defaultFile = process.cwd() + '/monitor.log';
+      // get writable out stream, either to custom location or default
+      const out = _finalOptions.monitorout
+        ? getCustomOut(_finalOptions.monitorout)
+        : createOutFileStream(defaultFile);
+
+      // callback with error if resolving the file path errored
+      if (out instanceof Error) return callback(out);
```

Comment:
I think we should prefer to wrap the `resolveMonitoringOutputFile` (or whatever its called) in a `try/catch` than depend on magic error return types. Less magic === less potential errors IMO